money.go

```diff
@@ -85,6 +88,33 @@ func New(amount int64, code string) *Money {
 	}
 }
 
+// NewFromFloat creates and returns new instance of Money from a float64.
+func NewFromFloat(amount float64, currency string) *Money {
+	currencyDecimals := math.Pow(10, float64(GetCurrency(currency).Fraction))
+	return New(int64(amount*currencyDecimals), currency)
+}
+
+// NewFromString creates and returns new instance of Money from a string.
+func NewFromString(amount string, currency string) (*Money, error) {
+	amount = strings.Trim(amount, "0")
+
+	var exponent int
+	if pointIndex := strings.Index(amount, "."); pointIndex != -1 {
+		exponent = len(amount[pointIndex+1:])
+	}
+
+	intString := strings.ReplaceAll(amount, ".", "")
+	parsed, err := strconv.ParseInt(intString, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("can't parse '%s' to money", amount)
+	}
+
+	currencyFraction := GetCurrency(currency).Fraction
+	parsed /= int64(math.Pow(10, float64(exponent-currencyFraction)))
```

Comment:
The use of floating point math can introduce rounding problems. Money uses int math just to avoid that.
What do you think using an alternate approach to parsing it¡
For example:
```go
func NewFromStr(amount string, currencyCode string) (*Money, error) {
	currency := m.GetCurrency(currencyCode)

	amountInt, err := parsers.ParseNumber(amount, currency.Fraction, currency.Decimal)

	if err != nil {
		 fmt.Errorf("can't parse '%s' to money", amount)
	}

	return New(amountInt, currencyCode)
}

```

Where:
```go
// ParseNumber parses the string for the representation of a number in
// format "dddd.dd" or "dddddd" (without decimal digit)
// Note: If excess decimals contains invalid characters, they are ignored
func ParseNumber(s string, decimals int, decimal string) (int64, error) {
	dotPos := strings.IndexByte(s, decimal)

	var toParse string

	var digits int

	if dotPos < 0 {
		toParse = s
	} else {
		digits = len(s) - dotPos - 1
		if digits > decimals {
			digits = decimals
		}

                // Remove decimal separator and in excess decimals
		toParse = s[:dotPos] + s[dotPos+1:dotPos+1+digits]
	}

	value, err := strconv.ParseInt(toParse, 10, 64)

	if err != nil {
		return 0, err
	}

        // Add missing decimal positions
	for d := digits; d < decimals; d++ {
		value *= 10
	}

	return value, nil
}

```

Reply:
Thank you for the suggestions, this is much better than my parser